dbt_coves/tasks/generate/sources.py

```diff
@@ -226,20 +292,231 @@ def get_templates_context(self, relation, columns, json_cols=None):
+        self.update_source_tables(source_a.get("tables"), source_b.get("tables"))
+
+    def merge_sources(self, sources_a, sources_b):
+        sources_a_names = [source.get("name") for source in sources_a]
+        for new_source in sources_b:
+            if not new_source.get("name") in sources_a_names:
+                sources_a.append(new_source)
+            else:
+                for current_source in sources_a:
+                    if current_source.get("name") == new_source.get("name"):
+                        self.update_sources_properties(current_source, new_source)
+        return sources_a
+
+    def merge_property_files(self, dict_a, dict_b):
+        result_dict = copy.deepcopy(dict_a)
+        if "sources" in result_dict:
+            result_dict["sources"] = self.merge_sources(
+                result_dict["sources"], dict_b["sources"]
+            )
+        if "models" in result_dict:
+            result_dict["models"] = self.merge_models(
+                result_dict["models"], dict_b["models"]
+            )
+
+        return result_dict
+
+    def update_property_file(self, template, context, yml_path, templates_folder):
+        # Grab and open current yml dict
+        current_yml = dict()
+        new_yml = dict()
+        with open(yml_path, "r") as file:
+            current_yml = yaml.safe_load(file)
+
+        # Render new one
+        render_template_file(
+            template,
+            context,
+            yml_path,
+            templates_folder=templates_folder,
+        )
+
+        # Open it
+        with open(yml_path, "r") as file:
+            new_yml = yaml.safe_load(file)
+
+        # Deep merge with old content
+        merged_ymls = self.merge_property_files(current_yml, new_yml)
+
+        with open(yml_path, "w") as file:
+            yaml.safe_dump(merged_ymls, file, sort_keys=False)
+
+    def render_property_file(self, template, context, model_yml, templates_folder):
+        model_yml.parent.mkdir(parents=True, exist_ok=True)
         render_template_file(
-            "source_model_props.yml",
+            template,
             context,
-            str(destination).replace(".sql", ".yml"),
+            model_yml,
             templates_folder=templates_folder,
         )
 
+    def render_property_files(
+        self, context, options, templates_folder, update_strategy, object
+    ):
+        strategy_key_update_all = ""
+        strategy_key_recreate_all = ""
+
+        if object == "Models":
+            template = "source_model_props.yml"
+            yml_cfg_destination = self.get_config_value("model_props_destination")
+            # If destination does not contain Jinja syntax it's a single file,
+            # and update/recreate must behave as 'update all' or 'recreate all'
+            if not re.search(
+                r"\{\{[A-Za-z]*\}\}", yml_cfg_destination.replace(" ", "")
+            ):
+                options["model_prop_is_single_file"] = "Yes"
```

Comment:
Why using strings instead of booleans?

Reply:
@ssassi because already-existent options used "Yes" and "No" as values, should I adapt everything to booleans?

Reply:
for example line 261

https://github.com/datacoves/dbt-coves/blob/831fff88bf599819185f5c5d88b05a67f561cef4/dbt_coves/tasks/generate/sources.py#L261

Reply:
The purpose of having those options was to manage three-state variables: `None`, set by user on `Yes`, or set by user on `No`.
If it was `None`, a question was asked to the user, if different from `None`, the value chosen by the user is remembered for future questions.
Are you using that same approach? (maybe it worth adding this comment I just posted here in above the dictionary for future reference).

Reply:
You are right

In my "new options", values are only set to "Yes" when using 'recreate all' or 'update all' (there is no three-state, just default and True)

I may default original values to `False` instead of `None`, and changing to `True` when the user opts to use update/recreate `all`